src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -759,6 +766,39 @@ PurgeConnectionForPlacement(ShardPlacement *placement)
 
 
 /*
+ * Removes a given connection from the transaction participant hash, based on
+ * the host and port of the provided connection. If the hash is not NULL, it
+ * MUST contain the provided connection, or a FATAL error is raised.
+ */
```

Comment:
Eh, passing on it. Don't want to iterate any more on this, as it's hopefully a stopgap regardless.